# 3. Asking meaningful question

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data_path = './Data/CarInformation.csv'
df = pd.read_csv(data_path)

| **Question**                                                | **Meaning of the Question**                                                |
|-------------------------------------------------------------|---------------------------------------------------------------------------|
| Which car is the most ideal?                                | Identify the relationships affecting the car's speed to choose the most ideal car. |
| How have production trends changed? Analyze these trends.    | Analyze the changes in car production trends over time.                   |
| Which origin is ideal for buying a car by mpg?                     | Identify the origin where car manufacturers have made the most improvements to buy durable and long-lasting cars by mpg. |



## 3.1. Question 1: Which car is the most ideal? 

Nowadays, young people are passionate about speed. Therefore, as part of the younger generation, we will focus on relationships surrounding speed to evaluate the vehicles. Here, we will examine which columns influence the speed column by using the correlation between them.

First, we will use the `corr` function to understand the relationships between them.

In [3]:
correlation_matrix = df[['weight', 'acceleration', 'mpg', 'horsepower']].corr()
print("Correlation coefficient:")
print(correlation_matrix)

Correlation coefficient:
                weight  acceleration       mpg  horsepower
weight        1.000000     -0.417457 -0.831741    0.864538
acceleration -0.417457      1.000000  0.420289   -0.689196
mpg          -0.831741      0.420289  1.000000   -0.778427
horsepower    0.864538     -0.689196 -0.778427    1.000000


After that, we will normalize the data to make it easier to calculate scores.

In [4]:
mpg = df['mpg']
mpg_norm = (mpg - mpg.min()) / (mpg.max() - mpg.min())

weight = df['weight']
weight_norm = (weight - weight.min()) / (weight.max() - weight.min())

acceleration = df['acceleration']
acceleration_norm = (acceleration - acceleration.min()) / (acceleration.max() - acceleration.min())

horsepower = df['horsepower']
horsepower_norm = (horsepower - horsepower.min()) / (horsepower.max() - horsepower.min())

Next, we will filter out the cars with the highest scores.

In [5]:
score = mpg_norm - weight_norm + acceleration_norm - horsepower_norm
df['score'] = score
score = score.sort_values(ascending=False)
df.loc[score.nlargest(10).index].reset_index(drop = True)

name   mpg  cylinders  displacement  horsepower  \
0                        vw pickup  44.0          4          97.0        52.0   
1               vw dasher (diesel)  43.4          4          90.0        48.0   
2             vw rabbit c (diesel)  44.3          4          90.0        48.0   
3  volkswagen rabbit custom diesel  43.1          4          90.0        48.0   
4                        mazda glc  46.6          4          86.0        65.0   
5                       datsun 210  40.8          4          85.0        65.0   
6            toyota corolla tercel  38.1          4          89.0        60.0   
7               chevrolet chevette  29.0          4          85.0        52.0   
8                   toyota starlet  39.1          4          79.0        58.0   
9                    renault 5 gtl  36.0          4          79.0        58.0   

   weight  acceleration  model_year  origin     score  
0    2130          24.6          82  europe  1.739754  
1    2335          23.7          80  europe  1.633841  
2    2085          21.7          80  europe  1.609612  
3    1985          21.5          78  europe  1.594145  
4    2110          17.9          80   japan  1.345112  
5    2110          19.2          80   japan  1.268238  
6    1968          18.8          80   japan  1.240054  
7    2035          22.2          76     usa  1.224896  
8    1755          16.9          81   japan  1.224816  
9    1825          18.6          77  europe  1.223712

- Top Performers: The top-scoring vehicles are the VW Pickup (1.739754) and VW Dasher (Diesel) (1.633841). This implies that these vehicles have a combination of attributes that contribute to a high score.

- Origin and Score: It appears that European and Japanese vehicles tend to have higher scores compared to American vehicles. This could suggest that fuel efficiency and other factors considered in the scoring metric are generally better in these regions.  